tests/ssr.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect } from 'react'
+import { act, screen } from '@testing-library/react'
+import { renderToString } from 'react-dom/server'
+import { describe, expect, it } from 'vitest'
+import { create } from 'zustand'
+
+const describeTest = React.version.startsWith('18') ? describe : describe.skip
+
+describeTest('ssr behavior with react 18', () => {
+  interface BearStoreState {
```

Comment:
```suggestion
  const { hydrateRoot } = await vi.importActual('react-dom/client')

  interface BearStoreState {
```

Reply:
I want to use it like this, but if I change it to "skipIf" and it runs, it throws an error because it doesn't actually skip the context.
```
describe.skipIf(!React.version.startsWith('18'))(
  'ssr behavior with react 18',
  async () => {
    const {
      hydrateRoot,
    }: {
      hydrateRoot: (
        container: Element | Document,
        initialChildren: React.ReactNode
      ) => void
    } = await vi.importActual('react-dom/client')
```
throw error with a message `Error: Failed to load url react-dom/client (resolved id: react-dom/client). Does the file exist?`

If I move the other context inside the it context, it resolves the issue.

I've applied the changes with it. 
Please take a look and feel free to share any further suggestions or improvements

Reply:
Yeah I mean to the top of the context